### PR TITLE
add registry that reports to SpectatorD

### PR DESCRIPTION
**Pull request by brharrington**
With this implementation Java would be consistent
with other languages based on thin clients.

### CODE REVIEW

File: spectator-reg-sidecar/src/main/java/com/netflix/spectator/sidecar/CommonTags.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  /**
+   * Extract common infrastructure tags from the Netflix environment variables.
+   *
+   * @param getenv
+   *     Function used to retrieve the value of an environment variable.
+   * @return
+   *     Common tags based on the environment.
+   */
+  static Map<String, String> commonTags(Function<String, String> getenv) {
```
Comment (copperlight) [inline spectator-reg-sidecar/src/main/java/com/netflix/spectator/sidecar/CommonTags.java:39]: Confirmed that these are the two we need to support for multi-container.

**Review by copperlight — APPROVED** [review]
This looks good! Very exciting to see this happen for the reference Java client.

File: spectator-reg-sidecar/src/test/java/com/netflix/spectator/sidecar/SidecarConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class SidecarConfigTest {
```
Comment (copperlight) [inline spectator-reg-sidecar/src/test/java/com/netflix/spectator/sidecar/SidecarConfigTest.java:26]: I noticed this earlier, but forgot to leave a note - was this test suite left empty intentionally?

Comment (brharrington) [reply spectator-reg-sidecar/src/test/java/com/netflix/spectator/sidecar/SidecarConfigTest.java:26]: Addressed in #967

**Pull request merged**